pkg/devfile/validate/components.go

```diff
@@ -4,12 +4,16 @@ import (
 	"fmt"
 
 	"github.com/openshift/odo/pkg/devfile/parser/data/common"
+	"github.com/openshift/odo/pkg/odo/util/pushtarget"
+	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 // Errors
 var (
-	ErrorNoComponents         = "no components present"
-	ErrorNoContainerComponent = fmt.Sprintf("odo requires atleast one component of type '%s' in devfile", common.ContainerComponentType)
+	ErrorNoComponents              = "no components present"
+	ErrorNoContainerComponent      = fmt.Sprintf("odo requires atleast one component of type '%s' in devfile", common.ContainerComponentType)
```

Comment:
can we have proper error types defined? instead of using variables
https://github.com/openshift/odo/pull/3549/files#diff-5fd81db0ba5aa2ae5579e01facb598e8R6

Reply:
so you're saying we should have a struct for each err types?

Reply:
updated - a1cd35d1e2dc20ca87dae6f5389177ddc4ea5e5c